docs/command-line-slangc.md

```diff
@@ -149,6 +149,68 @@ For completeness, here are the options that `slangc` currently accepts:
 
 * `-output-includes`: After pre-processing has been performed will output to via the diagnostics the hierarchy of paths to source files reached 
 
+* -Xname to specify arguments to downstream tool `name` (covered in more detail in "Downstream Arguments")
+
+### Downstream Arguments
+
+During a Slang compilation work may be performed by multiple other stages including downstream compilers and linkers. It isn't possible in general or perhaps even desirable to provide Slang command line equivalents of every option available at every stage of compilation. It is useful to be able to set options specific to a particular compilation stage - to alter code generation, linkage and other options.
+
+The mechanism used here is based on the `-X` mechanism used in GCC, to specify arguments to the linker.
+
+```
+-Xlinker option
+```
+
+When used, option is not interpreted by GCC, but is passed to the linker once compilation is complete. Slang extends this idea in several ways. First there are many more 'downstream' stages available to Slang. These different stages are known as `SlangPassThrough` types in the API and have the following names
+
+* `fxc` - FXC HLSL compiler
+* `dxc` - DXC HLSL compiler
+* `glslang` - GLSLANG GLSL compiler
+* `visualstudio` - Visual Studio C/C++ compiler
```

Comment:
I think some people would want to nit-pick the naming here, since Visual Studio is the IDE while the `cl.exe` compiler (which is developed by a distinct team) is often just "Visual C/C++." I'm not sure what is the ideal here, and we can probably keep `visualstudio` as an alias even if we add something that would be more "correct" according to the team that developers the compiler in question.

(This is one of those things like how there is no "DirectX 11" only "Direct3D 11" where I waste way too many brain cycles on trying to be accurate to the brand names of all these APIs and tools)